packages/dev/core/src/Engines/engine.ts

```diff
@@ -598,7 +598,7 @@ export class Engine extends ThinEngine {
 
             this._sharedInit(canvas, !!options.doNotHandleTouchAction, options.audioEngine!);
 
-            if (IsWindowObjectExist()) {
+            if (IsDocumentAvailable()) {

```

Comment:
We used to call `IsWindowObjectExist` but `IsDocumentAvailable` seems more appropriate (and that's what we test when we remove the listener)